src/systems/systems/src/hydraulic/brake_circuit.rs

```diff
@@ -322,6 +316,193 @@ impl SimulationElement for BrakeCircuit {
+        self.last_max_state = self.max_button.is_pressed();
+    }
+}
+
+pub struct AutobrakeDecelerationGovernor {
+    target: Acceleration,
+    i_gain: f64,
+    p_gain: f64,
+    last_error: f64,
+
+    current_output: f64,
+    acceleration: Acceleration,
+
+    is_engaged: bool,
+    time_engaged: Duration,
+    filter: f64,
+}
+impl Default for AutobrakeDecelerationGovernor {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+impl AutobrakeDecelerationGovernor {
+    // Low pass filter for controller acceleration input, time constant in second
+    const ACCELERATION_INPUT_FILTER: f64 = 0.15;
+
+    pub fn new() -> AutobrakeDecelerationGovernor {
+        Self {
+            target: Acceleration::new::<meter_per_second_squared>(10.),
+            i_gain: 0.02,
+            p_gain: 0.2,
+            last_error: 0.,
+
+            current_output: 0.,
+            acceleration: Acceleration::new::<meter_per_second_squared>(0.),
+            is_engaged: false,
+            time_engaged: Duration::from_secs(0),
+            filter: Self::ACCELERATION_INPUT_FILTER,
+        }
+    }
+
+    pub fn update_target(&mut self, deceleration_target: Acceleration) {
+        self.target = deceleration_target;
+    }
+
+    pub fn engage(&mut self) {
+        self.is_engaged = true;
+    }
+
+    pub fn is_engaged(&self) -> bool {
+        self.is_engaged
+    }
+
+    pub fn disable(&mut self) {
+        self.is_engaged = false;
+        self.time_engaged = Duration::from_secs(0);
+        self.target = Acceleration::new::<meter_per_second_squared>(10.);
+    }
+
+    pub fn time_engaged(&self) -> Duration {
+        self.time_engaged
+    }
+
+    pub fn is_on_target(&self, percent_margin_to_target: f64) -> bool {
+        self.is_engaged && self.acceleration < self.target * percent_margin_to_target / 100.
+    }
+
+    pub fn update(&mut self, context: &UpdateContext) {
+        let accel = context.long_accel();
+        self.acceleration = self.acceleration
+            + (accel - self.acceleration)
+                * (1. - std::f64::consts::E.powf(-context.delta_as_secs_f64() / self.filter));
+
+        if self.is_engaged {
+            self.time_engaged += context.delta();
+
+            let target_error = self.acceleration.get::<meter_per_second_squared>()
+                - self.target.get::<meter_per_second_squared>();
+
+            let p_term = self.p_gain * (target_error - self.last_error);
+            let i_term = self.i_gain * target_error;
+            self.current_output += p_term + i_term;
+
+            self.last_error = target_error;
+
+            self.current_output = self.current_output.min(1.).max(0.);
+        } else {
+            self.last_error = 0.;
+            self.current_output = 0.;
+        }
+    }
+
+    pub fn output(&self) -> f64 {
+        self.current_output
+    }
+
+    pub fn deceleration_value(&self) -> Acceleration {
+        self.acceleration
+    }
```

Comment:
**issue**: I'm assuming this returns e.g. `-2.0` for negative acceleration. As the function is currently named `deceleration_value` this would mean `-2.0` is _positive_ acceleration. Thus I'd consider the naming to be incorrect.

Another suggestion is to think about how to encapsulate this, and the usage, such that you don't have to expose it towards the `A320AutobrakeController` at all. I.e. maybe add a function here: `decelerating_at_or_above_rate(TARGET_TO_SHOW_DECEL_IN_MAX_MS2)`.